web/src/compose.js

```diff
@@ -64,7 +69,11 @@ export function show_preview_area() {
     $("#compose").addClass("preview_mode");
     $("#compose .preview_mode_disabled .compose_control_button").attr("tabindex", -1);
 
-    const content = $("textarea#compose-textarea").val();
+    let content = $("textarea#compose-textarea").val();
+    if (check_need_to_split_long_messages()) {
+        // add horizontal lines to indicate that the messages will be split
+        content = message_content_split_parts(content).join("\n---\n");
+    }
```

Comment:
This feels hacky; you should instead have `show_preview` and its caller `render_and_show_preview` be responsible for doing the manipulation on `content`, probably involving a loop over messages (and having dividers between messages that are rendered via `handlebars` and contain text like "Message 1" inside an `hrule`, or similar styling).